### ISSUE TITLE
Discover screen does not seem to start at the top with JAWS

### ISSUE THREAD

**Issue by KatieWoe** [phetsims/ratio-and-proportion#321]
**Test device**
Dell
**Operating System**
Win 10
**Browser**
Chrome
**Problem description**
For https://github.com/phetsims/QA/issues/582. Found with JAWS.
When entering the Discover screen from the home screen, the scene description seems to start in the middle, rather than from the top. I did not notice this on the Create screen. If I'm making some mistake myself, rather than this being a bug, feel free to close.

**Visuals**
https://drive.google.com/file/d/1PfHaN48osXy-_NhUw9XAdrG_cHZg7M3Q/view?usp=sharing

Troubleshooting information:

<details>

!!!!! DO NOT EDIT !!!!!
Name: ‪Ratio and Proportion‬
URL: https://phet-dev.colorado.edu/html/ratio-and-proportion/1.0.0-dev.77/phet/ratio-and-proportion_all_phet.html
Version: 1.0.0-dev.77 2020-12-08 00:32:10 UTC
Features missing: applicationcache, applicationcache, touch
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36
Language: en-US
Window: 1280x658
Pixel Ratio: 1.5/1
WebGL: WebGL 1.0 (OpenGL ES 2.0 Chromium)
GLSL: WebGL GLSL ES 1.0 (OpenGL ES GLSL ES 1.0 Chromium)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 30 uniform: 4096
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 32)
Max viewport: 32767x32767
OES_texture_float: true
Dependencies JSON: {}

</details>

**Comment by terracoda** [comment]
This issue may have been fixed with #282.
@KatieWoe, can you try again?

**Comment by KatieWoe** [comment]
This looks fixed on master. Tested on Windows 10 Chrome with Jaws.

**Comment by KatieWoe** [comment]
For https://github.com/phetsims/QA/issues/601, I am seeing this with both screens on NVDA.

https://user-images.githubusercontent.com/41024075/107091395-f530df00-67be-11eb-8a15-31840d24eeab.MOV


https://user-images.githubusercontent.com/41024075/107091595-480a9680-67bf-11eb-9768-b542c4d43bc7.MOV

**Comment by zepumph** [comment]
It seems like it is close to the top, but not quite. Why does it start on the second sentence of the second paragraph?!?!

![image](https://user-images.githubusercontent.com/6856943/107291052-06265e00-6a14-11eb-99a9-16f4d36e2340.png)

**Comment by zepumph** [comment]
@KatieWoe, I found similar behavior as you did here, likely because after pressing the button, we were both next pressing the down arrow, like for "next line". In this case, I believe that NVDA is focused on the sentence "The Play Area is a tall space. . . ", and by pressing the down arrow, we start at "You can move the Left . . . ". I am able to confirm this by looking at my laptop-specific key commands in https://www.nvaccess.org/files/nvda/documentation/userGuide.html:

>Read current line -> NVDA + L 



When I click on a home screen button, the press NVDA + L, the FIRST time I hear "out of list. the Play Area. . . ". Then I press NVDA + L again and only hear "The Play area . . ."

The only list I can think this applies to is from the home screen, so it seems like NVDA is getting confused about how we jump away from this list and focus the next screen view.

**Comment by zepumph** [comment]
I am feeling a bit stumped here. I looked into how the screen summary content was being set, as well as our order of operations around how we focus the screen view upon screenProperty change, and nothing seems obvious about this behavior. I'd like to touch base with @jessegreenberg at this point.

**Comment by terracoda** [comment]
I'm stumped, too. Last time I checked, it was working fine with VoiceOver. I double check again tomorrow.

**Comment by terracoda** [comment]
Yep, it works fine on VO, not sure why they same thing would work with JAWS.

**Comment by zepumph** [comment]
@jessegreenberg and I investigated heavily today with NVDA and chrome. The main issue we are trying to solve is why NVDA starts at the "Play area" paragraph instead of the first, intro paragraph. 

* We created an HTML scrap where a button focuses a div with two paragraphs. NVDA always auto reads to first paragraph on focus of the div.
* We removed the list structure from the Home screen to see if that was causing NVDA to read the play area paragraph. It wasn't.
* We moved the h1 to be inside the ScreenView primary sibling div to see if that would change the focus, it didn't.
* We directly focused the h1 when inside the ScreenView div, and that works. It seems like this bug has to do with when we directly call `focus` on the ScreenView.

**Comment by zepumph** [comment]
I feel like this could be a low level bug or implementation detail that @jessegreenberg and I don't fully understand. It would be good to get to the bottom of this.

**Comment by zepumph** [comment]
We feel like it may be good to pursue focusing directly on the h1, as that was the only thing that fixed our issue, but I'd like to sit on this for a bit.

**Comment by zepumph** [comment]
@jessegreenberg and I got together again today and decided to commit a change where we focus the h1 directly. It seemed like the best way forward, even though it worked around the fact that we never discovered why NVDA behaved so strangely when focusing directly on the ScreenView's primary sibling. @jessegreenberg will you please review this.

**Comment by jessegreenberg** [comment]
The new assertion is being hit on CT
> assert && assert( !this.pdomTitleNode.focusable, 'about to set to be focusable' );

I am not sure why yet, but I find that it happens quickly after clicking in the dev tools or the browser navigation bar, so when the document is no longer active.

**Comment by jessegreenberg** [comment]
Addition of pdomTitleNode in https://github.com/phetsims/joist/commit/5491e6a986d1dd22fe68fa9f154a4715236aafed looks good to me.

I can no longer hit the CT error after making the pdomTitleNode non-focusable when a screen becomes inactive. Indeed, I think the issue is that we don't get blur events when running with `?fuzz` while the sim is in the background. @zepumph can you please review https://github.com/phetsims/joist/commit/28e1fd7e56777efe4e24ad961a8615b939dc7734?

Alternatively, I couldn't remember why that assertion was important, could that have just been removed?

**Comment by jessegreenberg** [comment]
Sorry, https://github.com/phetsims/joist/commit/28e1fd7e56777efe4e24ad961a8615b939dc7734 wasn't right - I forgot that focus needs to change after the current screen becomes visible. Should be fixed in above commit.

**Comment by jessegreenberg** [comment]
There is another problem with this though, the AnimatedPanZoomListener assumed that the first focusable element had defined bounds. Zooming into the screen right on load while the h1 has focus will break the sim.

@zepumph when this SHA is cherry-picked into the release branch you will also need to include this patch to scenery https://github.com/phetsims/scenery/commit/f508d40fe2c202d78a3bd92bc37a7290a35c8182, which fixes this.

**Comment by zepumph** [comment]
Thanks for taking the lead at fixing my error. I like what you did. Upon seeing the order dependencies in that Sim function, and recognizing https://github.com/phetsims/joist/issues/208/https://github.com/phetsims/joist/issues/188, I felt like it would be best to try to keep this internal. Of course this means that there is a bit of an assumption about what having a visible ScreenView means (for the focusability of the title). This feels acceptable for me, please let me know if you disagree. I'm going to proceed with  another RC now. Here are the commits to grab in order:

https://github.com/phetsims/joist/commit/b579d6cd5d371d6f8798948497c73face1abc535
https://github.com/phetsims/joist/commit/5491e6a986d1dd22fe68fa9f154a4715236aafed
https://github.com/phetsims/joist/commit/4143de85a1f495c87f86dba0df1bda40d58138e7
https://github.com/phetsims/joist/commit/28e1fd7e56777efe4e24ad961a8615b939dc7734
https://github.com/phetsims/joist/commit/dca767dbeed6c4ceb1a7c7f2cc8bd0b467d59fd9
https://github.com/phetsims/scenery/commit/f508d40fe2c202d78a3bd92bc37a7290a35c8182
https://github.com/phetsims/joist/commit/a43be65c7a0ba7d9603998ab1bda86527a59ffd8

**Comment by KatieWoe** [comment]
Saw something odd if you enter from the home screen while reading through the description and not tabbing with keyboard nav. It seems to read a random part of the scene description, then start at the top. Seen with NVDA on firefox.

https://user-images.githubusercontent.com/41024075/108263484-930f9c80-7123-11eb-9d42-cb7196094cfe.MOV

**Comment by terracoda** [comment]
The "out of list" might just be what was cued up and left over from leaving the list of buttons on the home screen, and then I am not sure why the browser is announced, but it seems like it takes JAWS a few lines of code to get oriented on the new screen. 

It's good that JAWS eventually finds its way automatically to the top - or did you make it go there, @KatieWoe.

**Comment by KatieWoe** [comment]
>Seen with NVDA on firefox.

This was NVDA, I haven't checked it with Jaws yet. It did get there on it's own.

**Comment by zepumph** [comment]
I cannot reproduce this on Chrome + NVDA, but can for Firefox.

**Comment by zepumph** [comment]
Over the last 8 minutes, I have been able to reproduce this once. Only in Firefox on rc.2, never in master or or unbuilt mode with 1.0 shas checked out. I have also not been able to consistently get this behavior in rc.2. I don't know how to solve an issue like this when I can't reproduce it consistently. In the vast majority of cases, it focuses the h1 of the selected screen and says "Create Screen, Ratio and Proportion  heading level 1".

Part of me feels like we should not spend more time on this. What do you think @KatieWoe? If you can reproduce it easily and consistently, then I should most likely spend more time on it.

**Comment by KatieWoe** [comment]
I would say that, since it doesn't actually skip the top, it isn't as large a problem any more, though I did find it odd and distracting. I found it fairly easy to reproduce myself using the method described in https://github.com/phetsims/ratio-and-proportion/issues/321#issuecomment-780831469, but I can get more data and see if it is harder to replicate than I thought.

Also of note: NVDA is supported only on Firefox, and Jaws only on Chrome at the moment. We may check the other browser for more data on a particular issue, but tests are done only on those platform combos.

**Comment by zepumph** [comment]
Right. I am reproducing it more often now. I feel like this is the same behavior that @jessegreenberg and I couldn't figure out a fix for, so we decided to focus the heading instead of the whole div. It sounds exactly the same, with "out of list". I will double check with @jessegreenberg and likely not fix. It is weird and annoying, but it is nice that it ends up at the h1 at the end.

**Comment by zepumph** [comment]
@jessegreenberg and I feel like this is quite similar to the original problem, except that its behavior is much more usable. Given our investigation in https://github.com/phetsims/ratio-and-proportion/issues/321#issuecomment-776255039, and the fact that @jessegreenberg and I don't know what else we would try to get to the bottom of this. We are ready to close. We will add this to the screen ready bug tracking md file.
